src/cargo/util/toml/mod.rs

```diff
@@ -805,35 +1276,181 @@ pub struct TomlProject {
     #[serde(rename = "default-run")]
     default_run: Option<String>,
 
+    // Package metadata.
+    description: Option<MaybeWorkspace<String>>,
+    homepage: Option<MaybeWorkspace<String>>,
+    documentation: Option<MaybeWorkspace<String>>,
+    readme: Option<MaybeWorkspace<StringOrBool>>,
+    keywords: Option<MaybeWorkspace<Vec<String>>>,
+    categories: Option<MaybeWorkspace<Vec<String>>>,
+    license: Option<MaybeWorkspace<String>>,
+    #[serde(rename = "license-file")]
+    license_file: Option<MaybeWorkspace<String>>,
+    repository: Option<MaybeWorkspace<String>>,
+    metadata: Option<toml::Value>,
+    resolver: Option<String>,
+}
+
+#[derive(Clone, Debug)]
+struct DefinedTomlPackage {
```

Comment:
Is it possible to avoid duplicate type definitions like `TomlProject` and `DefinedTomlPackage`? I was hoping that we could get by where after some point in Cargo it's just statically known that the type here doesn't have `workspace = true` anywhere, and otherwise the "real" deserialized forms of things like `Manifest` and `Summary` in Cargo don't have to deal with the toml details at all.